src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1649,6 +1662,135 @@ def set_attention_slice(self, slice_size: Optional[int]):
         for module in modules:
             module.set_attention_slice(slice_size)
 
+    @classmethod
+    def from_pipe(cls, pipeline: "DiffusionPipeline", **kwargs) -> "DiffusionPipeline":
+        r"""
+        Create a new pipeline from a given pipeline. This method is useful to create a new pipeline with the same
+        weights and configurations without reallocating additional memory.
+
+        Arguments:
+            pipeline (`DiffusionPipeline`):
+                The pipeline from which to create a new pipeline.
+
+        Returns:
+            `DiffusionPipeline`:
+                A new pipeline with the same weights and configurations as `pipeline`.
+
+        Examples:
+
+        ```py
+        >>> from diffusers import StableDiffusionPipeline, StableDiffusionSAGPipeline
+
+        >>> pipe = StableDiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5")
+        >>> new_pipe = StableDiffusionSAGPipeline.from_pipe(pipe)
+        ```
+        """
+
+        pipeline_is_offloaded = True if hasattr(pipeline, "_all_hooks") and len(pipeline._all_hooks) > 0 else False
+        if pipeline_is_offloaded:
+            # `enable_model_cpu_offload` has be called on the pipeline, offload model and remove hook from model
```

Comment:
```suggestion
            # `enable_model_cpu_offload` has been called on the pipeline, offload model and remove hooks from model
```

Reply:
Why is this step needed?

Reply:
otherwise if you call `enable_model_cpu_offload` again on the new pipeline, it won't work correctly, 
however, this is no longer needed now we have this PR https://github.com/huggingface/diffusers/pull/7448
I can remove this now :)